server/routes/elasticsearch.ts

```diff
@@ -133,6 +136,87 @@ const getAliases = async (
   }
 };
 
+const createIndex = async (
+  req: Request,
+  h: ResponseToolkit,
+  callWithRequest: CallClusterWithRequest
+): Promise<ServerResponse<any>> => {
+  //@ts-ignore
+  const index = req.payload.indexConfig.index;
+  //@ts-ignore
+  const body = req.payload.indexConfig.body;
+  try {
+    await callWithRequest(req, 'indices.create', {
+      index: index,
+      body: body,
+    });
+  } catch (err) {
+    console.log('Anomaly detector - Unable to create index', err);
+    return { ok: false, error: err.message };
+  }
+  try {
+    const response: CatIndex[] = await callWithRequest(req, 'cat.indices', {
+      index,
+      format: 'json',
+      h: 'health,index',
+    });
+    return { ok: true, response: { indices: response } };
+  } catch (err) {
+    // In case no matching indices is found it throws an error.
+    if (
+      err.statusCode === 404 &&
+      get<string>(err, 'body.error.type', '') === 'index_not_found_exception'
+    ) {
+      return { ok: true, response: { indices: [] } };
```

Comment:
why return OK if indexIsNotFound? Seems like creating index doesn't work es expected in this case.

Reply:
yes, thanks for this. Was copying from existing code and this was leftover. This is wrong, will fix.